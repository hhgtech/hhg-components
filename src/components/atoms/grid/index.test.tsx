import React from 'react'
import { render, cleanup, screen } from '@testing-library/react'
import 'jest-styled-components'

import MockTheme from './../../../mockTheme'

import { Grid } from '.'

afterEach(cleanup)

test('it renders', () => {
  const grid = render(
    <MockTheme>
      <Grid columns={2} columnsDesktop={2}>
        <div>column 1</div>
        <div>column 2</div>
      </Grid>
    </MockTheme>,
  )

  const self = grid.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self.firstChild).toReturn
})
