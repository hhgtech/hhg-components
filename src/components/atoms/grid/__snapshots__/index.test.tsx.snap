// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  --column-count: 2;
  --columns-count-desktop: 2;
  --columns: var(--grid-template-columns,repeat(var(--columns-count),1fr));
  --columns-desktop: var( --grid-template-columns-desktop, repeat(var(--columns-count-desktop),1fr) );
  display: grid;
  grid-template-columns: var(--columns);
  grid-gap: var(--grid-gap);
  -webkit-align-items: var(--align-items);
  -webkit-box-align: var(--align-items);
  -ms-flex-align: var(--align-items);
  align-items: var(--align-items);
}

.c0 > * {
  justify-self: var(--justify-children);
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: var(--columns-desktop);
  }
}

<div
  class="c0"
>
  <div>
    column 1
  </div>
  <div>
    column 2
  </div>
</div>
`;
